### PR TITLE
Electron 6 Take Two

### PR DESCRIPTION
**Pull request by lkashef**
Original Issue: https://github.com/atom/atom/pull/20764

# TODO
- [x] Update the CI with the proper node and npm versions
- [x] Update electron version
- [x] Update electron dependencies (electron-chromedriver and electron-mksnapshot) to match the electron version
- [x] Get the app to bootstrap on the new electron version
- [x] Once the build job passes make sure the artifacts are runnable/installable
- [x] Fix any failing tests and make sure the test jobs are passing on the CI
- [x] Test the artifacts against the regression checklist below
- [x] Check performance issue after upgrade on Linux (Check https://github.com/atom/atom/issues/20924 and the conversation below)

# BLOCKERS AND ISSUES
- [x] fix scrollbar-style compilation problems with node 12.4.0
- [x] fix @atom/watcher compilation problems with node 12.4.0
- [x] Minimum supported macos version is 10.10 and not 10.9
  - [x] Update atom.io download page for beta and stable -- apparently nightly has been already updated ([PR](https://github.com/github/atom.io/pull/1376))
  - [x] Update flight manual ([PR](https://github.com/atom/flight-manual.atom.io/pull/634))
  - [x] Unblock flight manual CI ([PR](https://github.com/atom/flight-manual.atom.io/pull/635))
- [ ] Last release we agreed that we will not check for deprecation warnings, so I updated the template to reflect that (PR)
- [ ] `Linux: Atom scrolls even when not focused (#15482)` is a regression since electron 3, will update the template to keep to check during regression testing once it's fixed
- [x] It seems that `find-and-replace` in `find-in-project` is not working, even though tests are passing, check if others can reproduce this https://github.com/atom/find-and-replace/issues/1129
- [x] Settings page for packages doesn't scroll anymore  https://github.com/atom/atom/issues/21094
---

# Regression Checklist
## General testing

- [x] Review issues tagged with [`electron`](https://github.com/search?utf8=%E2%9C%93&q=org%3Aatom+label%3Aelectron+is%3Aopen+is%3Aissue) to see if any of them are fixed
  - [x] Fixes https://github.com/atom/atom/issues/967
  - [x] https://github.com/atom/atom/issues/6035 had already been fixed 
  - [x] https://github.com/atom/atom/issues/12935 had alreadybeen fixed
- [x] Review any comment in the Atom code that contains the keyword `TodoElectronIssue` to see if any of them can be resolved ([search link](https://github.com/search?q=org%3Aatom+TodoElectronIssue&type=Code)).
  - [x] The changes seems to be just a bit of code enhancement that I am not sure it's a top priority for now 
- [x] Verify that you can collaborate with Teletype
	- [x] Host using Atom's current stable release can collaborate with guest using Atom with the new version of Electron
	- [x] Guest using Atom's current stable release can collaborate with host using Atom with the new version of Electron
	- [x] Host and guest can collaborate when both are using Atom with the new version of Electron
  <details>
  <summary>Steps</summary>
	1. Open a file
	1. Share a portal
	1. Have a guest join the portal
	1. When guest edits the text in the file, verify that the host sees those changes
	</details>

### Text Input/Keybindings

See how to [setup keyboard layouts](https://github.com/atom/maintainers-log/blob/master/how-tos/setup-keyboard-layouts.md).

- [x] IME not working (https://github.com/atom/atom/issues/14911)
	- [x] macOS
	- [x] Windows 10
	- [x] Ubuntu
  <details>
  <summary>Repro steps</summary>

  1. Install an IME layout, for example Pinyin simplified  
  1. Open Atom and use the layout you installed as the keyboard layout
  1. Example, if you picked Pinyin simplified type in `zhongwenshuru`

  **Expected:** in Pinyin simplified `中文输入` is expected

  gif

  **Issues we've seen:** Only shows latin first character and not every character typed.

  gif
  </details>

- [x] macOS: Key binding with composition characters are ignored (https://github.com/atom/atom/issues/15189)
  <details>
  <summary>Repro steps</summary>

  1. Open your keymap.cson file and add the following:
      ```cson
      'atom-text-editor':
        # use hjkl; vim-like when pressing alt (word by word, paragraph by paragraph)
        # or alt (letter by letter)
        'alt-h': 'core:move-left'
        'alt-j': 'core:move-down'
        'alt-k': 'core:move-up'
        'alt-l': 'core:move-right'
      ```
  1. Save keymap.cson and reload atom.
  1. Try the new keymappings under `ABC - Extended` keyboard layout

  **Expected**: the keybindings work

  **Issues we've seen:** it types out the modified keys and ignores the key mapping
  </details>

- [x] macOS: Composition characters mess up insertion point (https://github.com/atom/atom/issues/15344)
  <details>
  <summary>Repro steps</summary>

  1. Open keymap.cson and add the following:
      ```cson
      'atom-text-editor':
        'alt-n': 'core:move-down'
        'alt-o': 'core:move-down'
      ```
  1. Have the following in your buffer:
      ```
      Foo
      Bar
      ```
  1. Have the cursor at the beginning of file(before F character) Type alt-n, then arrow right, then insert a character. Observe that the character is inserted at the beginning of the line (where the alt-n left you).
  1. Repeat steps using <kbd>alt-o</kbd> instead of <kbd>alt-n</kbd>.

  **Expected**: Both alt-o and alt-n behave the same

  **Issues we've seen:**
  bug

- [x] Ubuntu: Keystrokes involving <kbd>ctrl</kbd> resolve to the default layout instead of the active layout (https://github.com/atom/atom/issues/13170)

  <details>
  <summary>Repro steps</summary>

  **This should be tested on Linux with gnome**

  1. Install a Linux ditro with gnome, for example Ubuntu
  1. Install French AZERTY and English US QWERTY layouts
  1. Change the default layout to be French AZERTY
  1. Open Atom and switch to English US QWERTY layout
  1. Press ctrl-w

  **Expected**: `core:close` to dispatch (Or the keybinding-resolver to resolve to ctrl-w). Should be resolving to the keyboard layout that is chosen and not the OS default layout.

  **Issues we've seen:** `core:undo` dispatched because it was resolved as ctrl-z because AZERTY has Z where W is on QWERTY.
  </details>

- [x] Other keyboard layouts on new Electron version
  - [x] macOS: ABC - Extended
  - [x] macOS: Slovak
  - [x] macOS: U.S. International
  - [x] macOS: Squirrel (https://github.com/rime/squirrel)
  - [x] macOS: Full-Width Alphanumeric (Google) - Japanese layout  (https://www.google.co.jp/ime/)
  - [x] Windows: U.S.
  - [x] Linux: Pinyin Simplified

### UI

- [x] tree-view drag image (https://github.com/atom/tree-view/pull/1054)
	- [x] macOS
	- [x] Windows 10
	- [x] Ubuntu
  <details>
  <summary>Repro steps</summary>

  1. Open a project in Atom with multiple files and a few folders
  1. Drag a file into a folder

  **Expected:**  
  image

  **Issues we've seen:**

  image
  </details>

- [x] drag-and-drop indicator (https://github.com/atom/tree-view/pull/1055, https://github.com/atom/tabs/pull/426, https://github.com/atom/tabs/pull/437)
	- [x] macOS
	- [x] Windows 10
	- [x] Ubuntu
  <details>
  <summary>Repro steps</summary>

  1. Open Atom with a few folder projects on tree-view (you can do this on terminal in a directory that has existing folder and having a space between each folder. Example: `atom folderA folderB folderC`)
  1. Drag and drop all of the folders to reorder them on Atom

  **Expected:**
  The placeholder indicates where the folder/project is being dropped to.

  screen shot 2017-03-29 at 16 31 16

  **Issues we've seen:**
  No placeholder shows up after drag and dropping

  screen shot 2017-03-29 at 16 30 51
  </details>

- [x] Large file rendering (https://github.com/atom/atom/issues/16591)
	- [x] macOS
	- [x] Windows 10
	- [x] Ubuntu
  <details>
  <summary>Repro steps</summary>

  1. Open a file that has over one million lines
  1. Scroll down to the bottom

  **Expected:** To not regress the number of lines that can be rendered. Rendering to be correct for lines past a certain point. Atom 1.25  can render around 800k to 1 million lines correctly.

  **Issues we've seen:** Increased number of lines rendered but bad rendering past a certain point

  bad renderer
  </details>

- [x] Loss of subpixel AA when the cursor is at the end of long lines (https://github.com/atom/atom/issues/16889, https://github.com/atom/atom/pull/16595)
	- [x] macOS
	- [x] Windows 10
	- [x] Ubuntu
  <details>
  <summary>Repro steps</summary>

  1. Open [text-editor.js](https://github.com/atom/atom/blob/2e19cf4a7c0a4d624b1f9c1072cd292de112cdb0/src/text-editor.js) in Atom
  1. Copy the first 21 lines to a new untitled window
  1. Turn on soft wrapping and change language mode to javascript
  1. Resize the Window so it's soft wrapped
  1. On line 15 type `a` until the you reach the end of the window

  **Expected:** To not lose subpixel AA

  **Issues we've seen:** Loss of subpixel AA. Both when soft wrapping was [enabled](https://github.com/atom/atom/pull/16943) and [disabled](https://github.com/atom/atom/pull/16595).

  subpixel aa loss softwrap
  </details>

- [x] Scrolling horizontally shift + scroll wheel (https://github.com/atom/atom/pull/12696#issuecomment-288615912)
	- [x] macOS
	- [x] Windows 10
	- [x] Ubuntu
  <details>
  <summary>Repro steps</summary>

  1. Open a file that has long lines
  1. Turn off soft wrapping and resize the window so you have a horizontal scrollbar
  1. Press shift and scroll in both directions using the mouse wheel

  **Expected:** File to scroll horizontally in both directions

  **Issues we've seen:** File does not scroll horizontally at all in any direction
  </details>

- [x] Scrollbars misbehaving on the first file that is opened (https://github.com/atom/atom/pull/12696#issuecomment-251299649)
  <details>
  <summary>Repro steps</summary>

  1. Open Atom
  1. Open some file that has content enough to have a vertical scrollbar
  1. Close Atom
  1. Open Atom and look at the vertical scrollbar

  **Expected:** Scrollbar to be visible

  **Issues we've seen:** Scrollbar is not visible and is flickering when you are editing
  </details>

- [x] Middle clicking on unsaved tab (https://github.com/atom/atom/issues/15197)
	- [x] macOS
	- [x] Ubuntu
  <details>
  <summary>Repro steps</summary>

  1. Open Atom and open a new unsaved file.
  1. Make edits to the unsaved file and middle click on the tab.

  **Expected:**
  Clicking save/cancel or the options on the dialog works.

  **Issues we've seen:**
  The UI and the dialog is unable to receive mouse clicks. You can still choose options via Keyboard, but not mouse.
  </details>

- [x] Linux: Atom scrolls even when not focused (https://github.com/atom/atom/issues/15482)

  <details>
  <summary>Repro steps</summary>

  1. Open Atom with a large enough file that has a scrollbar
  1. Open a different application and place it over the Atom window
  1. Scroll with the mouse inside the other window
  1. Focus the Atom window

  **Expected:** Atom window should keep the original scroll position

  **Issues we've seen:** The Atom window scrolls after it is focused
  Note: this is a regression that has not been fixed since Electron 3, will leave it in the template to keep it on our radar once it's fixed 
  </details>

### Other

- [x] Supported Versions of OS (https://github.com/atom/atom/issues/15297)
	- [x] macOS
	- [x] Windows 10
	- [x] Ubuntu
  <details>
  <summary>Repro steps</summary>

  1. Check the list of supported platforms [Electron Master](https://github.com/electron/electron/blob/master/docs/tutorial/support.md#supported-platforms)
  1. Check what version is listed as minimum for each platform on https://atom.io and https://atom.io/faq

  **Expected:** https://atom.io and https://atom.io/faq to display the same versions as the Electron documentation

  **Issues we've seen:** macOS version support changed and https://atom.io and https://atom.io/faq was not updated
  </details>

- [x] macOS : Native tabs (https://github.com/atom/atom/pull/14711)
  <details>
  <summary>Repro steps</summary>

  1. Open two atom windows on mac
  1. Go to View in menus and check `Show Tab Bar`
  screen shot 2017-06-03 at 5 10 32 pm
  1. Do this for both Atom windows.
  1. Drag one atom window to the other via tab bar

  **Expected:**
  Native Tab bar shows up and able to drag different windows to each other.  Also make sure UI isn't messed up.

  **Issues we've seen:**
  Native Tab Bar doesn't work and messes up UI when enabled
  </details>

- [x] Ubuntu with KDE: Menu uses 100% CPU (https://github.com/electron/electron/issues/8455, https://github.com/atom/atom/issues/13885)
  <details>
  <summary>Repro steps</summary>

  1. Install Atom on Ubuntu KDE. [[instructions](https://linuxconfig.org/how-to-install-kde-plasma-desktop-on-ubuntu-18-04-bionic-beaver-linux)]
  1. Enable KDE plasma global menu: Right-click on the desktop -> Add Panel => Application Menu Bar. [[demo](https://youtu.be/qo5fhiXAUbI)]
  1. Start Atom.
  1. Try to use the desktop menu.

  **Expected:**
  Desktop menu items work

  **Issues we've seen:**
  Desktop menu items do nothing when you click on things.
  </details>

- [x] macOS: Slovak QWERTZ (https://github.com/atom/atom-keymap/issues/223)
  <details>
  <summary>Repro steps</summary>

  1. Install Slovak keyboard layout on mac
  1. Open keyboard viewer to see the keys are are suppose to show up and open keyboard resolver on atom.
  1. hit `cmd+'`.

  **Expected:**  Resolves to `cmd+'` like it shows on keyboard viewer on mac.

  **Issues we've seen:** It resolves to `ctrl+§`
  </details>

- [x] macOS: IME jump (https://github.com/atom/atom/issues/15696)
  <details>
  <summary>Repro steps</summary>

  1. Install an IME layout, for example Chinese pinyin
  1. Type one character

  **Expected:**
  Expected the IME to be positioned at under the character you entered
  **Issues we've seen:** The IME window is in the top left corner. When you enter the second character it jumps to be positioned under the text
  </details>

### CODE REVIEW

**Comment by DeeDeeG** [comment]
Hello! I'm excited to see this PR up.

I saw the CI error has to do with a missing script dependency. So I ran `npm install` in the `script/` dir and got this diff to `package-lock.json`:

<details><summary>diff for package-lock.json (click to expand):</summary>

```diff
diff --git a/script/package-lock.json b/script/package-lock.json
index a91f4f035..5755542f6 100644
--- a/script/package-lock.json
+++ b/script/package-lock.json
@@ -1269,7 +1269,8 @@
     "boolean": {
       "version": "3.0.1",
       "resolved": "https://registry.npmjs.org/boolean/-/boolean-3.0.1.tgz",
-      "integrity": "sha512-HRZPIjPcbwAVQvOTxR4YE3o8Xs98NqbbL1iEZDCz7CL8ql0Lt5iOyJFxfnAB0oFs8Oh02F/lLlg30Mexv46LjA=="
+      "integrity": "sha512-HRZPIjPcbwAVQvOTxR4YE3o8Xs98NqbbL1iEZDCz7CL8ql0Lt5iOyJFxfnAB0oFs8Oh02F/lLlg30Mexv46LjA==",
+      "optional": true
     },
     "boxen": {
       "version": "1.3.0",
@@ -1516,6 +1517,22 @@
       "resolved": "https://registry.npmjs.org/camelcase/-/camelcase-1.2.1.tgz",
       "integrity": "sha1-m7UwTS4LVmmLLHWLCKPqqdqlijk="
     },
+    "camelcase-keys": {
+      "version": "2.1.0",
+      "resolved": "https://registry.npmjs.org/camelcase-keys/-/camelcase-keys-2.1.0.tgz",
+      "integrity": "sha1-MIvur/3ygRkFHvodkyITyRuPkuc=",
+      "requires": {
+        "camelcase": "^2.0.0",
+        "map-obj": "^1.0.0"
+      },
+      "dependencies": {
+        "camelcase": {
+          "version": "2.1.1",
+          "resolved": "https://registry.npmjs.org/camelcase/-/camelcase-2.1.1.tgz",
+          "integrity": "sha1-fB0W1nmhu+WcoCys7PsBHiAfWh8="
+        }
+      }
+    },
     "caniuse-lite": {
       "version": "1.0.30000865",
       "resolved": "https://registry.npmjs.org/caniuse-lite/-/caniuse-lite-1.0.30000865.tgz",
@@ -6022,6 +6039,23 @@
         "unist-util-visit": "^1.1.0"
       }
     },
+    "meow": {
+      "version": "3.7.0",
+      "resolved": "https://registry.npmjs.org/meow/-/meow-3.7.0.tgz",
+      "integrity": "sha1-cstmi0JSKCkKu/qFaJJYcwioAfs=",
+      "requires": {
+        "camelcase-keys": "^2.0.0",
+        "decamelize": "^1.1.2",
+        "loud-rejection": "^1.0.0",
+        "map-obj": "^1.0.1",
+        "minimist": "^1.1.3",
+        "normalize-package-data": "^2.3.4",
+        "object-assign": "^4.0.1",
+        "read-pkg-up": "^1.0.1",
+        "redent": "^1.0.0",
+        "trim-newlines": "^1.0.0"
+      }
+    },
     "merge2": {
       "version": "1.2.2",
       "resolved": "https://registry.npmjs.org/merge2/-/merge2-1.2.2.tgz",
@@ -6274,30 +6308,6 @@
         "find-parent-dir": "^0.3.0"
       }
     },
-    "nodeify": {
-      "version": "1.0.1",
-      "resolved": "https://registry.npmjs.org/nodeify/-/nodeify-1.0.1.tgz",
-      "integrity": "sha1-ZKtpp7268DzhB7TwM1yHwLnpGx0=",
-      "requires": {
-        "is-promise": "~1.0.0",
-        "promise": "~1.3.0"
-      },
-      "dependencies": {
-        "is-promise": {
-          "version": "1.0.1",
-          "resolved": "https://registry.npmjs.org/is-promise/-/is-promise-1.0.1.tgz",
-          "integrity": "sha1-MVc3YcBX4zwukaq56W2gjO++duU="
-        },
-        "promise": {
-          "version": "1.3.0",
-          "resolved": "https://registry.npmjs.org/promise/-/promise-1.3.0.tgz",
-          "integrity": "sha1-5cyaTIJ45GZP/twBx9qEhCsEAXU=",
-          "requires": {
-            "is-promise": "~1"
-          }
-        }
-      }
-    },
     "noop-logger": {
       "version": "0.1.1",
       "resolved": "https://registry.npmjs.org/noop-logger/-/noop-logger-0.1.1.tgz",
@@ -10707,6 +10717,33 @@
         }
       }
     },
+    "redent": {
+      "version": "1.0.0",
+      "resolved": "https://registry.npmjs.org/redent/-/redent-1.0.0.tgz",
+      "integrity": "sha1-z5Fqsf1fHxbfsggi3W7H9zDCr94=",
+      "requires": {
+        "indent-string": "^2.1.0",
+        "strip-indent": "^1.0.1"
+      },
+      "dependencies": {
+        "indent-string": {
+          "version": "2.1.0",
+          "resolved": "https://registry.npmjs.org/indent-string/-/indent-string-2.1.0.tgz",
+          "integrity": "sha1-ji1INIdCEhtKghi3oTfppSBJ3IA=",
+          "requires": {
+            "repeating": "^2.0.0"
+          }
+        },
+        "repeating": {
+          "version": "2.0.1",
+          "resolved": "https://registry.npmjs.org/repeating/-/repeating-2.0.1.tgz",
+          "integrity": "sha1-UhTFOpJtNVJwdSf7q0FdvAjQbdo=",
+          "requires": {
+            "is-finite": "^1.0.0"
+          }
+        }
+      }
+    },
     "regenerate": {
       "version": "1.4.0",
       "resolved": "https://registry.npmjs.org/regenerate/-/regenerate-1.4.0.tgz",
@@ -11579,6 +11616,14 @@
       "resolved": "https://registry.npmjs.org/strip-eof/-/strip-eof-1.0.0.tgz",
       "integrity": "sha1-u0P/VZim6wXYm1n80SnJgzE2Br8="
     },
+    "strip-indent": {
+      "version": "1.0.1",
+      "resolved": "https://registry.npmjs.org/strip-indent/-/strip-indent-1.0.1.tgz",
+      "integrity": "sha1-DHlipq3vp7vUrDZkYKY4VSrhoKI=",
+      "requires": {
+        "get-stdin": "^4.0.1"
+      }
+    },
     "strip-json-comments": {
       "version": "1.0.4",
       "resolved": "https://registry.npmjs.org/strip-json-comments/-/strip-json-comments-1.0.4.tgz",
@@ -12249,25 +12294,11 @@
       "resolved": "https://registry.npmjs.org/throat/-/throat-2.0.2.tgz",
       "integrity": "sha1-qfzoCLaeEzpjJZB4DzQsMKYkmwI="
     },
-    "throttleit": {
-      "version": "0.0.2",
-      "resolved": "https://registry.npmjs.org/throttleit/-/throttleit-0.0.2.tgz",
-      "integrity": "sha1-z+34jmDADdlpe2H90qg0OptoDq8="
-    },
     "through": {
       "version": "2.3.8",
       "resolved": "https://registry.npmjs.org/through/-/through-2.3.8.tgz",
       "integrity": "sha1-DdTJ/6q8NXlgsbckEV1+Doai4fU="
     },
-    "through2": {
-      "version": "0.2.3",
-      "resolved": "https://registry.npmjs.org/through2/-/through2-0.2.3.tgz",
-      "integrity": "sha1-6zKE2k6jEbbMis42U3SKUqvyWj8=",
-      "requires": {
-        "readable-stream": "~1.1.9",
-        "xtend": "~2.1.1"
-      }
-    },
     "timed-out": {
       "version": "4.0.1",
       "resolved": "https://registry.npmjs.org/timed-out/-/timed-out-4.0.1.tgz",
@@ -12402,6 +12433,11 @@
       "resolved": "https://registry.npmjs.org/trim/-/trim-0.0.1.tgz",
       "integrity": "sha1-WFhUf2spB1fulczMZm+1AITEYN0="
     },
+    "trim-newlines": {
+      "version": "1.0.0",
+      "resolved": "https://registry.npmjs.org/trim-newlines/-/trim-newlines-1.0.0.tgz",
+      "integrity": "sha1-WIeWa7WCpFA6QetST301ARgVphM="
+    },
     "trim-repeated": {
       "version": "1.0.0",
       "resolved": "https://registry.npmjs.org/trim-repeated/-/trim-repeated-1.0.0.tgz",

```

</details>

Indeed the `meow` package was missing in a certain spot, as the CI error message indicates.

File: script/vsts/platforms/macos.yml
Comment (DeeDeeG) [inline script/vsts/platforms/macos.yml:25]: Probably want to un-comment this to re-enable the bootstrap cache on macOS CI.

_(I'm not sure if that would interrupt the current CI run if pushed right now, though!)_

**Comment by the-j0k3r** [comment]
See https://github.com/atom/atom/pull/20764#issuecomment-637084750

essentially also fixes #20819

Tried submitting fix upstream to electron https://github.com/electron/electron/pull/23881 but its been rejected since 5.x isnt supported. Though Fix has been backported to 6.X, 7.X and maybe others.

File: script/vsts/platforms/macos.yml
Comment (lkashef) [reply script/vsts/platforms/macos.yml:25]: O thanks for the heads up! I will do that on my end, no worries

**Comment by DeeDeeG** [comment]
The `macOS test packages1` job is bootstrapping for an unusually long time. I think it might be stuck and won't finish.

https://github.visualstudio.com/Atom/_build/results?buildId=76015&view=logs&j=6de74252-975b-522d-7a3d-227ca79ca6c1

**Comment by lkashef** [comment]
> An error occurred while provisioning resources (Error Type: Disconnect).

@DeeDeeG the error sound like a problem with VSTS, re-running the CI

**Comment by DeeDeeG** [comment]
Some tests timed out on Windows x64.

**Comment by lkashef** [comment]
Might have a release blocker

# 7/17/2020

> - [ ] It seems that `find-and-replace` in `find-in-project` is not working, even though tests are passing, check if others can reproduce this

**Comment by the-j0k3r** [comment]
> Might have a release blocker

How well are the electron bumps tested? Are the plugins shipped with Atom tested?

**Comment by lkashef** [comment]
@the-j0k3r Core packages are tested against atom and find and replace feature tests have actually passed on the CI, something might be wrong with my environment, I am still checking but shared this comment to get feedback if anyone wanted to volunteer to double check this (CI artifacts are public)

**Comment by Aerijo** [comment]
I can't check right now, but do those tests include packages like `snippets`? The CI is consistently failing recently, and best I can tell it was triggered by an Electron upgrade https://github.com/atom/snippets/pull/307.

**Comment by DeeDeeG** [comment]
I can reproduce the problem.

#### Steps to Reproduce:

1. Download and install atom-amd64.deb from the CI artifacts here: https://github.visualstudio.com/9e67709c-1fe5-47f2-8cf6-4902878a11f5/_apis/build/builds/76025/artifacts?artifactName=atom-amd64.deb&api-version=5.1&%24format=zip
1. Run `atom-dev`
1. Have some project open, or add one ([File -> [Add Project Folder...])
1. From the Menu bar: [Find] -> [Find in Project]
1. Type something to find next to the [Find All] button, press the [Find All] button
1. Type something to replace it with next to the [Replace All] button, press the [Replace All] button
1. App asks for confirmation: "Are you sure you want to replace all?"
   1. Press "OK" to confirm.

#### Expected:

Text should be replaced.

#### Actual:

Nothing happens after confirming/pressing "OK" to replace text.

This is a regression compared to the Nightly builds.

**Comment by aminya** [comment]
This might be due to using an old ripgrep with does the regex operation in Find and Replace. Could you try with this pull request merged?
https://github.com/atom/atom/pull/21056

**Comment by aminya** [comment]
Other than find and replace the issue, I have another problem. 
- Go to settings > packages
- Open the settings view for one package
- Try to scroll down. 

It does not work for me. I have to use the tab key to go through the settings

**Comment by aminya** [comment]
Thank you for the recent updates. Both find and replace and scrolling in the settings view issues are gone.

However, now I cannot search for packages for downloading in the settings view. Does this happen for others or it is just me? I built this on a branch that had other changes, so I am not certain if it is caused by those two updates.

image

**Comment by darangi** [comment]
> Thank you for the recent updates. Both find and replace and scrolling in the settings view issues are gone.
> 
> I tested this with the artifacts from @DeeDeeG. This hidden issue does not happen on those. Maybe something is wrong with my working branch.

Yea, I also confirmed it works fine on this build. Thanks for the effort.

**Review by aminya — APPROVED** [review]
Thank you so much for this work.

**Comment by aminya** [comment]
Let us know if there is anything blocking this from merging. We may be able to help.

File: script/vsts/platforms/linux.yml
Comment (DeeDeeG) [inline script/vsts/platforms/linux.yml:30]: Hi all.

In my opinion it could be good to have _newer_ `npm` in CI, rather than pinning to `npm@6.9.0`.

The system- wide npm is only used to install `script/package.json`, after which point the `script/node_modules/.bin/npm` is used to install `apm`, and `apm` is used to install the main packages.

System-wide npm is also used to install CI dependencies (`script/vsts/node_modules` and things like that). Again, here too it would be nice to have recent npm for speed, up-to-date dependency tree algorithm, and various bug-fixes.

In my view, just having an up-to-date npm is the priority. As various small bugs have been fixed over time. Some of the old bugs are inconvenient for things we are trying to do at our fork, so we would possibly patch the CI configs at our fork to have recent `npm`. Also: old `npm` (older than 6.12.0) doesn't work with Python 3. So at least that version would be nice. (Apparently all the CI environments have Python 2, so it's not a huge deal.)

Comment (DeeDeeG) [reply script/vsts/platforms/linux.yml:30]: Similar thought about Node 12. Newer is mostly better, although 12.16.x is the newest one can go in 12.x without running into a problem: https://github.com/atom/atom/issues/21091

Comment (aminya) [reply script/vsts/platforms/linux.yml:30]: I would say let's merge this as it is, and implement your suggestions in separate pull requests. This is well tested by us as of now.

**Comment by aminya** [comment]
Could you check my comment at https://github.com/atom/apm/pull/889#issuecomment-671160524

I am seeing some issues with `apm`. The Node version difference makes the native modules that are compiled using apm unusable. I want to know if others experience that too.

**Comment by jeff-hykin** [comment]
On a similar note @aminya , could the version of node being used be added to the package.json dependencies?  (E.g. npm install -s node)

The atom build fails with newer versions of node because of native modules, but the version of node isn't mentioned in the documentation or repo AFAIK

**Comment by aminya** [comment]
> On a similar note @aminya , could the version of node being used be added to the package.json dependencies? (E.g. npm install -s node)
> 
> The atom build fails with newer versions of node because of native modules, but the version of node isn't mentioned in the documentation or repo AFAIK

Yes, we should unify the Node versions identical and write it down in the `package.json`. 

If for some reason, the old Electron was not picky about the Node versions, the new one is, and we should absolutely make this unification before merging this pull request.

**Comment by aminya** [comment]
@darangi apm 2.5.1 solves the issue for Electron 5. You should merge this as well: https://github.com/atom/apm/pull/896/

**Comment by the-j0k3r** [comment]
Congrats on the merge =) and thanks for the work on these electron bumps

**Comment by the-j0k3r** [comment]
Please dont forget https://github.com/atom/atom/issues/20819 is fixed with electron 6.x merged

**Pull request merged**
